### ISSUE TITLE
Why catch unhandled rejections?

### ISSUE THREAD

**Issue by arcanis** [kripken/emscripten#7855]
This is a problem I always eventually have when working with Emscripten, so I figured I should open an issue because at the moment it makes for painful debugging sessions and error reports if you forget this detail.

Why is Emscripten's default shell configured to automatically catch unhandled rejections and exit the program? Shouldn't that be a decision taken by the user of the Emscriptened code, rather than Emscripten itself? This is even more problematic when Emscripten is used to compile a library, because then it silently bleeds into the code of the consumer applications - and goes unnoticed until it actually matters.

In particular, this problem causes stacktraces to disappear and to be replaced by an unhelpful generic error message (since Node doesn't print the exception if it sees nothing bound to `unhandledRejection`).

cc @dschuff @kripken 
ref #5948 / #7135

**Comment by kripken** [comment]
Yeah, I'd love to remove this too. Running our test suite to remember why we have it:

 * on node 8.12 an unhandled rejection at hello world startup (say, invalid wasm binary contents due to  them being corrupted somehow) prints out the error, then a warning about an unhandled rejection, a deprecation notice, then hangs for a few seconds, then exits with returncode 0.
 * node node 10.15, the error is logged, it hangs, then it prints the warning and deprecation etc, then exits with returncode 0.

So in both cases an error is handled very awkwardly, and worst no error code is emitted, so tests might miss stuff (which happens on the wasm waterfall, 2 unexpected successes).

On the other hand, I don't understand the issue you're seeing, since I **do** see stack traces emitted, while you say they disappear. Do you have a testcase I can see that with?

**Comment by kripken** [comment]
(and in the emscripten test suite, `other.test_only_force_stdlibs` fails also for an unexpected success)

**Comment by roccomuso** [comment]
is there a flag to remove it?

**Comment by kripken** [comment]
#9028 is open to add a flag for this.
